### DISCUSSION TITLE
Dynamic generation of classes will not work with JIT

### DISCUSSION THREAD

**Discussion by NeurAlch** [tailwindlabs/tailwindcss#6763] [Help] [answered]
I had different issues with Tailwind CSS 3, but found a new one today, I solved it with a dummy div/classes, wonder if there is another way to do this?

```
<div className={`__tw_text-${fontSize}`}>
  <div className={'__tw_text-2xl __tw_text-xl __tw_text-lg __tw_text-sm'} />
  {value}
</div>
```

The issue here is that `fontSize` is dynamic, so it never get compiled to classes, so the JIT never sees and generates those, and that happens a lot with dynamic code and styling.

**I love the JIT and the power it has**, plus the new ideas it brings, but I had looked a few times for alternatives to Tailwind in the last month, even tho I'm an early supporter and would not like to switch. Maybe we need a small base.css for tailwind?

**Comment by NeurAlch** [comment, marked as answer]
So far using dummy `className` seems to be the best way to include dynamic classes generated by code.

```<div className={'__tw_text-2xl __tw_text-xl __tw_text-lg __tw_text-sm'} />```

**Comment by Americoas** [comment]
Same issue still no fix to this.

[Liteblue]( https://www.liteblue.biz/)

**Comment by adamwathan** [comment]
It’s never been possible to use dynamic class names with Tailwind and still purge the output — this is explained in the docs for v3 here:

https://tailwindcss.com/docs/content-configuration#dynamic-class-names

But it’s also in the docs for v2.

**Reply by NeurAlch** [reply]
I think I probably explained it incorrectly, what I meant was that I would like to have a base of those classes like in Tailwind 2 (text-2xl, etc) so that dynamic classes are not an issue, since those already exist in the css file. 

In my case I don't have a problem with a few mb files of CSS since I'm building an extension that is downloaded only once. Is there a way to get all the css classes listed somehow from tailwind 2 so I can force version 3 to generate them?

**Comment by dapobelieve** [comment]
I'm having this issue too,
`const computedColors = computed(() =>
Object.values({
  text: 'text-${data.theme[props.trend]}-600',
  bg: 'bg-${data.theme[props.trend]}-100'
})
)`

if the classes I want to dynamically apply have never been used anywhere in my code, they wont be found when this code computes

**Reply by wongjn** [reply]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Comment by mybearworld** [comment]
If you really need to, use the `safelist` option in the config.
